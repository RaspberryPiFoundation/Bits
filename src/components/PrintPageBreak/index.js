import { Base } from 'react-iotacss'
import classnames from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'

/**
 * Places a component, hidden from display which forces a page break before it when printing
 */
export const PrintPageBreak = ({ children, className, ...props }) => {
  const classNames = classnames(className, 'c-print-page-break')

  return <Base className={classNames} {...props} />
}

PrintPageBreak.propTypes = {
  className: PropTypes.string,
}

export default PrintPageBreak
