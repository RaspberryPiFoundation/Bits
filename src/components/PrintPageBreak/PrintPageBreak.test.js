import { shallow } from 'enzyme'
import PrintPageBreak from '../PrintPageBreak'
import React from 'react'

let printPageBreak

const className = 'some-extra-class-name'

describe('<PrintPageBreak />', () => {
  describe('Renders...', () => {
    beforeEach(() => {
      printPageBreak = shallow(<PrintPageBreak />)
    })

    it('without crashing', () => {
      expect(printPageBreak.find('.c-print-page-break')).toExist()
    })

    it('matches its snapshot', () => {
      expect(printPageBreak).toMatchSnapshot()
    })
  })

  describe('With classNames', () => {
    it('adds the correct className', () => {
      printPageBreak = shallow(<PrintPageBreak className={className} />)
      expect(printPageBreak.find('.c-print-page-break')).toHaveClassName(
        className
      )
    })
  })
})
