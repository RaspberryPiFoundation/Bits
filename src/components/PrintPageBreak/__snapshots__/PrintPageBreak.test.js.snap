// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrintPageBreak /> Renders... matches its snapshot 1`] = `
<Base
  className="c-print-page-break"
  tagName="div"
/>
`;
